doc/howtos/rdtraining/04_basicmodel.rst

```diff
@@ -0,0 +1,298 @@
+    for example.
+
+When the fields are added to the model, restart the server with ``-u estate``
+
+.. code-block:: console
+
+    $ ./odoo-bin --addons-path=../custom,../enterprise/,addons -d rd-demo -u estate
+
+Connect to ``psql`` and check the structure of the table ``estate_property``. You'll notice that
+a couple of of extra fields were also added to the table. We will come back to them later.
+
+Common Attributes
+-----------------
+
+.. note::
+
+    **Goal**: at the end of this section, the fields ``name`` and ``expected_price`` should be
+    not nullable in the table ``estate_property``:
+
+    .. code-block:: console
+
+        rd-demo=# \d estate_property;
+                                                    Table "public.estate_property"
+            Column       |            Type             | Collation | Nullable |                   Default
+        --------------------+-----------------------------+-----------+----------+---------------------------------------------
+        ...
+        name               | character varying           |           | not null |
+        ...
+        expected_price     | double precision            |           | not null |
+        ...
+
+Much like the model itself, its fields can be configured by passing
+configuration attributes as parameters::
+
+    name = field.Char(required=True)
+
+Some attributes are available on all fields, here are the most common ones:
+
+:attr:`~odoo.fields.Field.string` (``str``, default: field's name)
+    The label of the field in UI (visible by users).
+:attr:`~odoo.fields.Field.required` (``bool``, default: ``False``)
+    If ``True``, the field can not be empty, it must either have a default
+    value or always be given a value when creating a record.
+:attr:`~odoo.fields.Field.help` (``str``, default: ``''``)
+    Long-form, provides a help tooltip to users in the UI.
+:attr:`~odoo.fields.Field.index` (``bool``, default: ``False``)
+    Requests that Odoo create a `database index`_ on the column.
+
+.. exercise:: Set attributes to existing fields.
+
+    Add the following attributes:
+
+    ========================= =========================
+    Field                     Attribute
+    ========================= =========================
+    name                      required
+    expected_price            required
+    ========================= =========================
+
+After restarting the server, both fields should be not nullable.
+
+Automatic Fields
+----------------
+
+**Reference**: the documentation related to this topic can be found in
+:ref:`reference/fields/automatic`.
+
+As noticed previously, extra fields were added to the table ``estate_property``.
+Odoo creates a few fields in all models\ [#autofields]_. These fields are
+managed by the system and shouldn't be written to. They can be read if
+useful or necessary:
```

Comment:
Should we specify that these fields are not meant to be used for business logic?